packages/discovery-provider/src/tasks/entity_manager/entities/dashboard_wallet_user.py

```diff
@@ -0,0 +1,224 @@
+            )
+        try:
+            signature_address = get_address_from_signature(metadata["wallet_signature"])
+        except:
+            raise IndexingValidationError(
+                "Invalid Create Dashboard Wallet User Transaction, signature provided is invalid"
+            )
+        if not signature_address or not signature_address.lower() == dashboard_wallet:
+            raise IndexingValidationError(
+                "Invalid Create Dashboard Wallet User Transaction, signature provided is invalid"
+            )
+    else:
+        raise IndexingValidationError(
+            f"Invalid Dashboard Wallet User Transaction, action {params.action} is not valid"
+        )
+
+
+def create_dashboard_wallet_user(params: ManageEntityParameters):
+    metadata = get_create_dashboard_wallet_user_metadata_from_raw(params.metadata)
+    if not metadata:
+        raise IndexingValidationError(
+            "Invalid Dashboard Wallet User Transaction, unable to parse metadata"
+        )
+    validate_dashboard_wallet_user_tx(params, metadata)
+    user_id = params.user_id
+    dashboard_wallet = metadata["wallet"]
+
+    dashboard_wallet_user_record = DashboardWalletUser(
+        user_id=user_id,
+        wallet=cast(
+            str, dashboard_wallet
+        ),  # cast to assert non null (since we validated above)
+        txhash=params.txhash,
+        blockhash=params.event_blockhash,
+        blocknumber=params.block_number,
+        updated_at=params.block_datetime,
+        created_at=params.block_datetime,
+    )
+
+    validate_dashboard_wallet_user_record(dashboard_wallet_user_record)
+    params.add_record(dashboard_wallet, dashboard_wallet_user_record)
+    return dashboard_wallet_user_record
+
+
+def delete_dashboard_wallet_user(params: ManageEntityParameters):
+    metadata = get_delete_dashboard_wallet_user_metadata_from_raw(params.metadata)
+    if not metadata:
+        raise IndexingValidationError(
+            "Invalid Dashboard Wallet User Transaction, unable to parse metadata"
+        )
+    validate_dashboard_wallet_user_tx(params, {})
+    dashboard_wallet = metadata["wallet"]
+    existing_dashboard_wallet_user = params.existing_records["DashboardWalletUser"][dashboard_wallet]
+    if dashboard_wallet in params.new_records["DashboardWalletUser"]:
+        existing_dashboard_wallet_user = params.new_records["DashboardWalletUser"][dashboard_wallet][-1]
+
+    deleted_dashboard_wallet_user = copy_record(
+        existing_dashboard_wallet_user,
+        params.block_number,
+        params.event_blockhash,
+        params.txhash,
+        params.block_datetime,
+    )
+
+    delete_dashboard_wallet_user.is_delete = True
+
+    validate_dashboard_wallet_user_record(deleted_dashboard_wallet_user)
+    params.add_record(dashboard_wallet, deleted_dashboard_wallet_user)
+    return deleted_dashboard_wallet_user
+
+
+def validate_dashboard_wallet_user_record(dashboard_wallet_user):
```

Comment:
nit , but should we move this function above create like the other helper functions are?